### PR TITLE
fix: add runtime version in build

### PR DESCRIPTION
**Pull request by smile37773**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/spring-cloud/azext_spring_cloud/_buildservices_factory.py [hunk 1]
```diff
@@ -56,14 +57,20 @@ def _get_upload_info(self):
...
-    def _queue_build(self, relative_path=None, builder=None, target_module=None, app=None, **_):
+    def _queue_build(self, relative_path=None, builder=None, target_module=None, runtime_version=None, app=None, **_):
         subscription = get_subscription_id(self.cmd.cli_ctx)
         service_resource_id = '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.AppPlatform/Spring/{}'.format(subscription, self.resource_group, self.service)
         properties = models.BuildProperties(
             builder='{}/buildservices/default/builders/{}'.format(service_resource_id, builder),
             agent_pool='{}/buildservices/default/agentPools/default'.format(service_resource_id),
             relative_path=relative_path,
             env={"BP_MAVEN_BUILT_MODULE": target_module} if target_module else None)
+        if(runtime_version == SupportedRuntimeValue.JAVA8):
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/_buildservices_factory.py:68]: If your target module is `None`, the env.add will throw NPE

**Review by yuwzho — CHANGES_REQUESTED** [review]

**Comment by yonzhan** [comment]
spring cloud

File: src/spring-cloud/azext_spring_cloud/_buildservices_factory.py (hunk 1 above)
Comment (smile37773) [reply src/spring-cloud/azext_spring_cloud/_buildservices_factory.py:68]: OK. fixed.

File: src/spring-cloud/azext_spring_cloud/_buildservices_factory.py [hunk 2]
```diff
@@ -74,6 +75,25 @@ def _queue_build(self, relative_path=None, builder=None, target_module=None, app
...
+            else:
+                env = {"BP_MAVEN_BUILT_MODULE": target_module}
+        else:
+            if(runtime_version == SupportedRuntimeValue.JAVA8):
+                env = {"BP_JVM_VERSION": "8.*"}
+            elif(runtime_version == SupportedRuntimeValue.JAVA11):
+                env = {"BP_JVM_VERSION": "11.*"}
+            elif(runtime_version == SupportedRuntimeValue.JAVA17):
+                env = {"BP_JVM_VERSION": "17.*"}
+        return env
```
Comment (yuwzho) [inline src/spring-cloud/azext_spring_cloud/_buildservices_factory.py:95]:
```suggestion
    def _get_build_env(self, target_module, runtime_version):
        if all(x is None for x in [target_module, runtime_version]):
            return None
        env = {}
        if target_module:
            env['BP_MAVEN_BUILT_MODULE'] = target_module
        
        runtime_version_table = {
            SupportedRuntimeValue.JAVA8: '8.*',
            SupportedRuntimeValue.JAVA11: '11.*',
            SupportedRuntimeValue.JAVA17: '17.*'
        }
        if runtime_version:
            env['BP_JVM_VERSION'] = runtime_version_table.get(runtime_version, '11.*')
        return env
```

**Review by yuwzho — APPROVED** [review]

**Comment by yuwzho** [comment]
#4294

**Review by wangzelin007 — APPROVED** [review]

**Pull request merged**